### ISSUE TITLE
aws s3 sync status takes time to calculate total/remaining files

### ISSUE THREAD

**Issue by yoni-yad2** [aws/aws-cli#7957]
### Describe the bug

Running `aws s3 sync [s3 folder with 10 files] [local folder] `
and for half of the runtime the status doesn't show real numbers of total/remaining files

Only after a certain amount of time it changes to reflect the real numbers.

### Expected Behavior

Status shown at the start:
Completed xxx GiB/~yyy GiB (zzz MiB/s) with **~10** file(s) remaining **(10 files)**

+ it will be nice to have the ETA shown as `Estimated time remaining: [dd] days [hh] hours [mi] minutes [ss] seconds`

### Current Behavior

While it copies the first 5 files in parallel the status shown is:
Completed xxx GiB/~yyy GiB (zzz MiB/s) with **~5** file(s) remaining **(calculating...)**

### Reproduction Steps

Upload 10 big files (10 GB each) to an s3 folder and run from and ec2 VM:
`aws s3 sync [s3 folder] [local folder]`

### Possible Solution

`aws s3 sync` should try first scanning for a few seconds all files/subfolders and calculate total number of files and size to quickly display it in the command status.

If scanning takes more than a few seconds then it should start syncing the files/subfolders and display `with total (calculating...) file(s) remaining (calculating...)` and eventually display the real total/remaining number after if finished scanning the entire folder tree.

### CLI version used

2.11.25

### Environment details (OS name and version, etc.)

Windows Server 2022 Datacenter / Powershell

**Comment by vaibhav0077** [comment]
Hi @hssyoo,
Can I work on this task ?

**Comment by adev-code** [comment]
Hello @yoni-yad2, thanks for reaching out and brining up the idea! I will relay this to our team and see if we can consider it. I will update you as soon as I get any updates.

**Comment by adev-code** [comment]
As of this current version `aws-cli/2.34.48`, it only shows "calculating..."

Uploading:
```
$ aws s3 sync ~/s3sync-test/upload s3://bucket/sync-test/
Completed 8.9 GiB/~50.0 GiB (189.0 MiB/s) with ~5 file(s) remaining (calculating...)   
...
upload: ./bigfile_2.bin to s3://bucket/sync-test/bigfile_2.bin                   
upload: ./bigfile_10.bin to s3://bucket/sync-test/bigfile_10.bin               
upload: ./bigfile_3.bin to s3://bucket/sync-test/bigfile_3.bin
Completed 56.1 GiB/100.0 GiB (181.4 MiB/s) with 6 file(s) remaining
```
Donwloading: 
```
downloading:
$ aws s3 sync s3://bucket/sync-test/ ~/s3sync-test/download/
Completed 1.2 GiB/~50.0 GiB (184.1 MiB/s) with ~5 file(s) remaining (calculating...)   
...
Completed 26.2 GiB/~50.0 GiB (192.2 MiB/s) with ~5 file(s) remaining (calculating...)  
...
download: s3://bucket/sync-test/bigfile_1.bin to ../download/bigfile_1.bin       
download: s3://bucket/sync-test/bigfile_4.bin to ../download/bigfile_4.bin     
download: s3://bucket/sync-test/bigfile_3.bin to ../download/bigfile_3.bin     
download: s3://bucket/sync-test/bigfile_2.bin to ../download/bigfile_2.bin
download: s3://bucket/sync-test/bigfile_10.bin to ../download/bigfile_10.bin
Completed 58.6 GiB/100.0 GiB (181.9 MiB/s) with 5 file(s) remaining
```

The team has considered this feature request. Leaving this FR issue open for tracking.
